module/VuFind/src/VuFind/RecordDriver/DefaultRecord.php

```diff
@@ -734,9 +734,9 @@ protected function getOpenUrlFormat()
         $formats = $this->getFormats();
         if (in_array('Book', $formats) || in_array('eBook', $formats)) {
             return 'Book';
-        } elseif (in_array('Article', $formats)) {
+        } elseif (in_array('Article', $formats) || in_array('electronic Article', $formats)) {
```

Comment:
Rather than adding more hard-coded values here, what if you add new class properties for `$journalFormats` and `$articleFormats` and then do an `array_intersect` to see if they are present? This way, we can leave the default values at the current `['Journal']` and `['Article']`, but it's very easy to adjust in a local subclass for other situations (or we might even be able to make it configurable somewhere).

I would prefer not to add new format strings here that are not found in the language files or anywhere else, as I think that could lead to confusion... though if you think that's the only way forward, I'm certainly willing to discuss further!

Reply:
You are right. Hardcoded values are a quick'n'dirty solution. 

It'll think about it, there might be a longer development arc: It should be possible to separate the serialization of a record (OpenURL, Schema.org, XML, Citations) from the RecordDriver.